File: 0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        int min = prices[0];
        int max=0;
        int pro=0;
        for(int i=1;i<n;i++){
            min=fmin(prices[i],min);
            pro=prices[i]-min;
            max=fmax(pro,max);
        }
        return max;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int max = 0;
        int n =prices.length;
        for(int i=0;i<n;i++){
            int cur = prices[i];
            if(cur<min){
                min = cur;
            }else {
                max = Math.max(max,cur-min);
            }
        }
        return max;
    }
}
